vizro-core/src/vizro/models/_navigation/nav_item.py

```diff
@@ -0,0 +1,97 @@
+from __future__ import annotations
+
+import itertools
+from typing import Optional
+
+import dash
+import dash_bootstrap_components as dbc
+from dash import html
+from pydantic import validator
+
+from vizro._constants import STATIC_URL_PREFIX
+from vizro.models import VizroBaseModel
+from vizro.models._models_utils import _log_call
+from vizro.models._navigation._navigation_utils import _validate_pages
+from vizro.models._navigation.accordion import Accordion
+from vizro.models.types import NavPagesType
+
+
+class NavItem(VizroBaseModel):
+    """Icon to be used in Navigation Panel of Dashboard.
+
+    Args:
+        tooltip (Optional[str]): Text to be displayed in the tooltip on icon hover.
+        icon (Optional[str]): URI (relative or absolute) of the embeddable content.
+        pages (NavPagesType): See [NavPagesType][vizro.models.types.NavPagesType].
+    """
+
+    tooltip: Optional[str]
+    icon: Optional[str]
+    pages: NavPagesType
+    selector: Optional[Accordion] = None
+    text: Optional[str] = ""
+    max_text_length: int = 9
+
+    # Re-used validators
+    _validate_pages = validator("pages", allow_reuse=True, always=True)(_validate_pages)
+
+    @validator("selector", pre=True, always=True)
+    def set_selector(cls, selector, values):
+        if selector is None:
+            return Accordion(pages=values.get("pages"))
+        return selector
+
+    @validator("icon", always=True, pre=True)
+    def set_icon(cls, icon):
+        if icon is None:
+            return STATIC_URL_PREFIX + "/images/icon_1.svg"
+        return icon if icon else None
```

Comment:
As mentioned, can we check whether we can read this in from the google material icon library instead of having static images in our package. Optimally the user should be able to consume any icon from the google icon library via 

`icon = "some url to the icon"`

Reply:
Implemented by Li.